brightwind/analyse/analyse.py

```diff
@@ -2245,3 +2252,295 @@ def calc_air_density(temperature: Union[float, pd.Series, pd.DataFrame],
+
+            air_density = (P_d / (R_d * T)) + (P_v / (R_v * T))
+
+        where:
+            P_d = dry air partial pressure (Pa) = total pressure - vapour pressure  
+            P_v = vapour pressure (Pa)  
+            R_d = specific gas constant for dry air (287.05 J/kg*K)  
+            R_v = specific gas constant for water vapour (461.5 J/kg*K)  
+            T   = air temperature in Kelvin (°C + 273.15)
+
+    **Example usage**
+    ::
+    import brightwind as bw
+
+    vapour_pressure = 12.3   # hPa
+    pressure = 1013.25       # hPa
+    temperature = 15.0       # °C
+
+    air_density = bw.calc_air_density_from_vapour_prs(vapour_pressure, pressure, temperature)
+    print(air_density)
+    # 1.219391176982024
+
+    # calculate air density from air temperature, dew point temperature and air pressure:
+    T_air = 15
+    T_dew_pt = 14
+    P_air = 1000
+    vapour_prs = bw.calc_vapour_pressure_from_dewpoint(T_dew_pt)
+    air_density = bw.calc_air_density_from_vapour_prs(vapour_prs, P_air, T_air)
+    air_density
+    # 1.2016918568274164 
+
+    # calculate air density from air temperature, relative humidity and air pressure:
+    T_air = 15
+    P_air = 1000
+    RH = 50
+    vapour_prs = bw.calc_vapour_pressure_from_relative_humidity(RH, T_air)
+    air_density = bw.calc_air_density_from_vapour_prs(vapour_prs, P_air, T_air)
+    air_density
+    # 1.2050986677333038
+
+    # calculate air density from air temperature, relative humidity and air pressure series:
+    DATA = bw.load_csv(bw.demo_datasets.demo_data)
+    DATA = bw.apply_cleaning(DATA, bw.demo_datasets.demo_cleaning_file)
+
+    vapour_prs = bw.calc_vapour_pressure_from_relative_humidity(DATA.RH2m, DATA.T2m)
+    air_density = bw.calc_air_density_from_vapour_prs(vapour_prs, DATA.P2m, DATA.T2m)
+    air_density.tail(5)
+    # Timestamp
+    # 2017-11-23 10:10:00    1.195418
+    # 2017-11-23 10:20:00    1.196145
+    # 2017-11-23 10:30:00    1.196055
+    # 2017-11-23 10:40:00    1.195684
+    # 2017-11-23 10:50:00    1.197327
+    # dtype: float64
+    """
+    # Check dimensions of temperature, pressure and rel_humidity_percent if not float or int
+    if isinstance(vapour_pressure_hPa, pd.Series) or (isinstance(air_pressure_hPa, pd.Series)) or (isinstance(air_temperature_degC, pd.Series)):
+        if len(vapour_pressure_hPa) != len(air_pressure_hPa):
+            raise ValueError("vapour_pressure_hPa and air_pressure_hPa must have the same dimensions.") 
+        if len(air_temperature_degC) != len(air_pressure_hPa):
+            raise ValueError("temperature, vapour_pressure_hPa and air_pressure_hPa must have the same dimensions.")
+
+    P_v_Pa = 100*vapour_pressure_hPa
+    P_Pa = 100*air_pressure_hPa
+    T_K = air_temperature_degC + 273.15
+    R_d = GAS_CONST_DRY_AIR
+    R_v = GAS_CONST_WATER
+    P_d_Pa = P_Pa - P_v_Pa
+    air_density = P_d_Pa/(R_d*T_K) + P_v_Pa/(R_v*T_K)
+    return air_density
+
+
```

Comment:
I would suggest to call function `calc_saturation_water_vapour_pressure `and reword a bit wording on docstring to follow this. 

I don't think that we should say that it used by windpro but if we want to say this then should say also on the main `calc_air_density_XXX`